public/app/plugins/datasource/cloudwatch/query-runner/CloudWatchMetricsQueryRunner.ts

```diff
@@ -172,21 +171,7 @@ export class CloudWatchMetricsQueryRunner extends CloudWatchRequest {
   }
 
   filterMetricQuery(query: CloudWatchMetricsQuery): boolean {
-    const { region, metricQueryType, metricEditorMode, expression, metricName, namespace, sqlExpression, statistic } =
-      query;
-    if (!region) {
-      return false;
-    }
-    if (metricQueryType === MetricQueryType.Search && metricEditorMode === MetricEditorMode.Builder) {
-      return !!namespace && !!metricName && !!statistic;
-    } else if (metricQueryType === MetricQueryType.Search && metricEditorMode === MetricEditorMode.Code) {
-      return !!expression;
-    } else if (metricQueryType === MetricQueryType.Query) {
-      // still TBD how to validate the visual query builder for SQL
-      return !!sqlExpression;
-    }
-
-    throw new Error('invalid metric editor mode');
+    return filterMetricsQuery(query);
```

Comment:
Making this static so it can be used in the tracking handler (tracking handler can't access the data source file).